reframe/core/logging.py

```diff
@@ -243,101 +207,88 @@ def _create_syslog_handler(handler_config):
     else:
         address = (host, port)
 
-    facility = handler_config.get('facility', 'user')
+    facility = site_config.get(f'{config_prefix}/facility')
     try:
         facility_type = logging.handlers.SysLogHandler.facility_names[facility]
     except KeyError:
-        raise ConfigError('syslog handler: '
-                          'unknown facility: %s' % facility) from None
+        # This should not happen
+        raise AssertionError(
+            f'syslog handler: unknown facility: {facility}') from None
 
-    socktype = handler_config.get('socktype', 'udp')
+    socktype = site_config.get(f'{config_prefix}/socktype')
     if socktype == 'udp':
         socket_type = socket.SOCK_DGRAM
     elif socktype == 'tcp':
         socket_type = socket.SOCK_STREAM
     else:
-        raise ConfigError('syslog handler: unknown socket type: %s' % socktype)
+        # This should not happen
+        raise AssertionError(
+            f'syslog handler: unknown socket type: {socktype}'
+        )
 
     return logging.handlers.SysLogHandler(address, facility_type, socket_type)
 
 
-def _create_stream_handler(handler_config):
-    stream = handler_config.get('name', 'stdout')
+def _create_stream_handler(site_config, config_prefix):
+    stream = site_config.get(f'{config_prefix}/name')
     if stream == 'stdout':
         return logging.StreamHandler(stream=sys.stdout)
     elif stream == 'stderr':
         return logging.StreamHandler(stream=sys.stderr)
     else:
-        raise ConfigError('unknown stream: %s' % stream)
+        # This should not happen
+        raise AssertionError(f'unknown stream: {stream}')
 
 
-def _create_graylog_handler(handler_config):
+def _create_graylog_handler(site_config, config_prefix):
     try:
         import pygelf
     except ImportError:
         return None
 
-    host = handler_config.get('host', None)
-    port = handler_config.get('port', None)
-    extras = handler_config.get('extras', None)
-    if host is None:
-        raise ConfigError('graylog handler: no host specified')
-
-    if port is None:
+    address = site_config.get(f'{config_prefix}/address')
+    host, *port = address.split(':', maxsplit=1)
```

Comment:
Why do you need `*port` since `maxsplit=1`?

Reply:
It is correct:

```pycon
>>> 'server:123'.split(':', maxsplit=1)
['server', '123']
```

Reply:
It is correct:

```pycon
>>> 'server:123'.split(':', maxsplit=1)
['server', '123']
```

Reply:
Yes, but why do you need the starred `port`?

Reply:
Yes, because the user may not specify the port and the Graylog handler needs it. We could enforce this at the schema level, but I just did as before.

Reply:
So if the user does not specify, you need the `port` to be an empty list?

Reply:
No, I need the port to be a number, but I use the `*` notation to avoid handling exceptions and be more concise. Later on, I do `port = port[0]`.

Reply:
The only case that I'm not handling and I should is when the input is `server:`.

Reply:
Ok, I get it now, thanks.